src/shelly_hap_rgb.cpp

```diff
@@ -212,43 +215,76 @@ void RGBWLight::HSVtoRGBW(const HSV &hsv, RGBW &rgbw) const {
   }
 }
 
-void RGBWLight::SetOutputState(const char *source) {
-  LOG(LL_INFO,
-      ("state: %s, brightness: %i, hue: %i, saturation: %i", OnOff(cfg_->state),
-       cfg_->brightness, cfg_->hue, cfg_->saturation));
+void RGBWLight::UpdateOnOff(bool on, const std::string &source) {
+  if (cfg_->state != on) {
```

Comment:
pattern: reduce indentation to improve readability. deal with rare / error condition first and bail , otherwise continue with the main flow at the same level on indent. makes code easier to follow.

```
void RGBWLight::UpdateOnOff(bool on, const std::string &source) {
  if (cfg_->state == on) return;
  // deal with other case
}
```